Framework/PythonInterface/test/python/plugins/algorithms/PeakMatchingTest.py

```diff
@@ -0,0 +1,305 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from  plugins.algorithms.PeakMatching import PeakMatching as _PeakMatching
+import unittest
+from unittest import mock
+from mantid.simpleapi import DeleteWorkspace,CreateEmptyTableWorkspace,PeakMatching,CreateWorkspace
+from mantid.api import mtd , ITableWorkspace
+
+
+class PeakMatchingTest(unittest.TestCase):
+
+    def setUp(self):
+        self.peak_data = {'Primary energy': {'Ag': {'K(4->1)': 3177.7,
+                                                    'L(4->2)': 900.7,
+                                                    'M(4->3)': 304.7,
+                                                    '6->5': 141.0}},
+
+                          'Secondary energy': {'Ag': {'K(2->1)': 3140.6,
+                                                      'L(8->2)': 1347.8,
+                                                      'M(10->3)': 567.0,
+                                                      '8->6': 122.2}},
+
+                          'All energies': {'Ag': {'K(4->1)': 3177.7,
+                                                  'L(4->2)': 900.7,
+                                                  'M(4->3)': 304.7,
+                                                  '6->5': 141.0,
+                                                  'K(2->1)': 3140.6,
+                                                  'L(8->2)': 1347.8,
+                                                  'M(10->3)': 567.0,
+                                                  '8->6': 122.2}}}
+
+        self.input_peaks = [(900, 0.8), (306, 0.8), (567, 0.8), (3, 0.8)]
+        self.input_table =CreateEmptyTableWorkspace(OutputWorkSpace="input")
+        self.input_table.addColumn("double","centre")
+        self.input_table.addColumn("double", "sigma")
+        for row in self.input_peaks:
+            self.input_table.addRow(row)
+        self.algo = _PeakMatching()
+
+    def tearDown(self):
+        self.delete_if_present("input")
+        self.delete_if_present("primary_matches")
+        self.delete_if_present("secondary_matches")
+        self.delete_if_present("all_matches")
+        self.delete_if_present("all_matches_sorted_by_energy")
+        self.delete_if_present("element_likelyhood")
+        self.delete_if_present("Test")
+        self.algo = None
+
+    def assertPeaksMatch(self,table,peaks):
+        assert(len(peaks) == table.columnCount())
+        for column in peaks:
+            column_data = table.column(column)
+            assert(len(peaks[column]) == len(column_data))
+            for i in range(len(peaks[column])):
+                if type(peaks[column][i]) == float:
+                    self.assertAlmostEqual(peaks[column][i],column_data[i])
```

Comment:
better using a delta then you know the tolerance on the equality condition

Also if there are no floats this will look like it passes